### PR TITLE
fix: Upgrade Gradle and dependencies to fix Java 20

### PR DESCRIPTION
**Pull request by Myself5**
* Add and configure [refreshVersions](https://jmfayard.github.io/refreshVersions/) module to update dependencies
* Update dependencies to latest version for Gradle compatibility
* Update Gradle to version 8.1.1

Note: Jacoco can not be automatically updated by the update and its version will need to be updated in the versions.properties manually.

Test: Build and run on ArchLinux using OpenJDK 20.0.1.

### CODE REVIEW

**Comment by andrii-bodnar** [comment]
Hi @Myself5, thanks a lot for your contribution! it looks interesting

There are failed CI/CD checks in this PR, could you please take a look? Thanks

**Comment by andrii-bodnar** [comment]
Also, I am wondering if it will be backward compatible with the previous CLI installations

**Comment by Myself5** [comment]
Hey @andrii-bodnar, working on the CI/CD issues.

I don't quite see how it would break any kind of previous CLI installations it only updates deps and the gradle tool similiar to what I did before, just that this time I added the [refreshVersions](https://jmfayard.github.io/refreshVersions/) module to make updating the dep-versions easier in the future.

File: build.gradle [hunk 1]
```diff
@@ -1,11 +1,12 @@
...
+import static de.fayard.refreshVersions.core.Versions.versionFor
 
 plugins {
     id 'java'
-    id 'com.github.johnrengelman.shadow' version '7.0.0'
+    id 'com.github.johnrengelman.shadow'
     id 'jacoco'
-    id "com.github.spotbugs" version "4.7.1"
-    id 'org.asciidoctor.jvm.convert' version '3.1.0'
+    id "com.github.spotbugs"
```
Comment (andrii-bodnar) [inline build.gradle]:
```suggestion
    id 'com.github.spotbugs'
```

**Review by andrii-bodnar — APPROVED** [review]
@Myself5 thank you!

**Pull request merged**